spec/unit/environments_spec.rb

```diff
@@ -90,6 +91,13 @@ module PuppetEnvironments
       end
     end
 
+    it "proceeds with non-existant env dir" do
```

Comment:
@joshcooper I added a test. lmk if there's a better way to describe this scenario 😅

Reply:
As long as the test fails with the previous version of `lib/puppet/environments.rb`, then LGTM

Reply:
it does 😊 
```
Failures:

  1) Puppet::Environments directories loader proceeds with non-existant env dir
     Failure/Error:
       valid_environment_names.collect do |name|
         create_environment(name)
       end

     NoMethodError:
       undefined method `collect' for nil:NilClass
```